OpenEXR/lib/IlmMacros.cmake

```diff
@@ -0,0 +1,52 @@
+# Means we don't need to set the openexr directory everywhere
+# Uses ${Package}_ROOT
+# Sets ${Package}_INCLUDE_DIR
+macro(_ILM_FIND_INCLUDE_DIR package header)
+	FIND_PATH(${package}_INCLUDE_DIR
+		NAMES ${header}
+		PATH_SUFFIXES include/OpenEXR
+		PATHS ${${package}_ROOT} $ENV{PROGRAMFILES}/openexr
+		)
+	mark_as_advanced(${package}_INCLUDE_DIR)
+endmacro()
+
+# Finds each specified library
+# Uses ${Package}_ROOT, ${Package}_USE_STATIC_LIBS
+# Sets ${Package}_LIBRARIES
+macro(_ILM_FIND_LIBRARIES package components)
+	string(TOLOWER ${package} lowerpackage)
+
+	# Adapted from FindBoost.cmake
+	# Support preference of static libs by looking for static libs first
+	if(${package}_USE_STATIC_LIBS)
+		set(_${lowerpackage}_ORIG_CMAKE_FIND_LIBRARY_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES})
+		if(WIN32)
+			set(CMAKE_FIND_LIBRARY_SUFFIXES .lib .a ${CMAKE_FIND_LIBRARY_SUFFIXES})
+		else()
+			set(CMAKE_FIND_LIBRARY_SUFFIXES .a)
+		endif()
+	endif()
+
+	set(_${lowerpackage}_LIB_VARS)
+	set(${package}_LIBRARIES)
+	foreach(COMPONENT ${_${lowerpackage}_COMPONENTS})
+		string(TOUPPER ${COMPONENT} UPPERCOMPONENT)
+
+		find_library(${package}_${UPPERCOMPONENT}_LIBRARY
+			NAMES ${COMPONENT}
+			PATHS ${${package}_ROOT}/lib $ENV{PROGRAMFILES}/openexr/lib
+			)
```

Comment:
The RHEL comment is a good point! I hadn't though about that so I'll definitely give that a go and test it in a VM.

On naming conventions, I'd always been unsure about it. I'd only ever found this mailing list message about it (https://cmake.org/pipermail/cmake/2009-March/027414.html) but your comment made me go and find an answer once and for all. According to the documentation here (https://cmake.org/cmake/help/v3.4/manual/cmake-developer.7.html#standard-variable-names) variables set by a Find module should use exact case (ie whatever case the module name uses), which are almost always camelcase. The standard modules are a bit mixed on this but I think that's for backwards compatibility. (eg FindBoost and FindGit are camelcase but FindJPEG and FindZLIB are uppercase). I'd be interested to know what other people think as well.